File: test/testspringboot/src/test/java/pers/acp/test/application/test/nobuild/RestTest.java
package pers.acp.test.application.test.nobuild;

import com.fasterxml.jackson.databind.JsonNode;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.http.*;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.util.MultiValueMap;
import pers.acp.test.application.BaseTest;

/**
 * @author zhangbin by 28/04/2018 15:54
 * @since JDK 11
 */
class RestTest extends BaseTest {

    @Test
    void testGet() {
        String name = "abc";
        String pwd = "qwertyyu";
        ResponseEntity<JsonNode> result = testRestTemplate.getForEntity("/boot/rest2/" + name + "?pwd={1}", JsonNode.class, pwd);
        Assertions.assertEquals(HttpStatus.OK, result.getStatusCode());
        Assertions.assertEquals(result.getBody().get("result2").textValue(), name + pwd);
    }

    @Test
    void testPost() {
        String request = "sdfasdafsfa范德萨范德萨";
        ResponseEntity<JsonNode> result = testRestTemplate.postForEntity("/boot/rest2", request, JsonNode.class);
        Assertions.assertEquals(HttpStatus.OK, result.getStatusCode());
        Assertions.assertEquals(result.getBody().get("result2").textValue(), request);
    }

    @Test
    void testFormPost() {
        HttpHeaders headers = new HttpHeaders();
        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
        MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
        params.add("name", "你好");
        params.add("number", "23");
        params.add("money", "23.98");
        HttpEntity<MultiValueMap<String, String>> requestEntity = new HttpEntity<>(params, headers);
        ResponseEntity<JsonNode> result = testRestTemplate.postForEntity("/boot/mapform", requestEntity, JsonNode.class);
        Assertions.assertEquals(HttpStatus.OK, result.getStatusCode());
        System.out.println(result.getBody().toString());
    }

}

File: test/testkotlin/src/test/kotlin/pers/acp/test/kotlin/test/nobuild/RestTest.kt
package pers.acp.test.kotlin.test.nobuild

import pers.acp.test.kotlin.component.BeanBeanResolver
import pers.acp.test.kotlin.entity.TableOne
import pers.acp.test.kotlin.repository.MemberRepository
import pers.acp.test.kotlin.repository.TableOneRepository
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.expression.spel.standard.SpelExpressionParser
import org.springframework.expression.spel.support.StandardEvaluationContext
import org.springframework.http.HttpStatus
import pers.acp.test.kotlin.test.TestData
import pers.acp.test.kotlin.BaseTest


/**
 * @author zhangbin by 28/04/2018 17:18
 * @since JDK 11
 */
internal class RestTest : BaseTest() {

    @Autowired
    lateinit var beanBeanResolver: BeanBeanResolver

    @Autowired
    lateinit var tableOneRepository: TableOneRepository

    @Autowired
    lateinit var memberRepository: MemberRepository

    @Test
    fun testQuery200() {
        val name = "abc"
        val result = testRestTemplate.getForEntity("/query?name={1}", TableOne::class.java, name)
        Assertions.assertEquals(HttpStatus.OK, result.statusCode)
        Assertions.assertEquals(result.body!!.name, name)
    }

    @Test
    fun testAdd201() {
        val name = "fsdafa"
        val tableOne = TableOne().apply {
            this.name = name
            this.value = 101.0
        }
        val result = testRestTemplate.postForEntity("/add", tableOne, TableOne::class.java)
        Assertions.assertEquals(HttpStatus.CREATED, result.statusCode)
        Assertions.assertEquals(result.body!!.name, name)
    }

    @Test
    fun testSpEL() {
        val parser = SpelExpressionParser()
        val context = StandardEvaluationContext()
        context.beanResolver = beanBeanResolver
        val testData = TestData(test2 = 100)
        val exp2 = parser.parseExpression("@testComponent.match(test2)")
        println("Expression Value: ${exp2.getValue(context, testData)}")
        println(testData)
    }

}